iroh-net/src/hp/portmapper/pcp/protocol/response.rs

```diff
@@ -197,4 +211,59 @@ impl Response {
             data,
         })
     }
+
+    #[cfg(test)]
+    fn random(opcode: Opcode) -> Self {
+        let data = OpcodeData::random(opcode);
+        Self {
+            lifetime_seconds: rand::random(),
```

Comment:
nit: I prefer using a passed in `Rng` for these, so you can use seeded randomness in tests (which is great when you have to debug these)